src/util/codec/table.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+use std::io::Write;
+
+use super::{number, Result, Error, Datum, datum};
+
+const ID_LEN: usize = 8;
+const PREFIX_LEN: usize = 1 + ID_LEN /*table_id*/ + 2;
+const RECORD_ROW_KEY_LEN: usize = PREFIX_LEN + ID_LEN;
+const TABLE_PREFIX: &'static [u8] = b"t";
+const RECORD_PREFIX_SEP: &'static [u8] = b"_r";
+const INDEX_PREFIX_SEP: &'static [u8] = b"_i";
+
+fn append_table_record_prefix(mut buf: &mut [u8], table_id: i64) -> Result<()> {
+    try!(buf.write_all(TABLE_PREFIX));
+    try!(number::encode_i64(buf, table_id));
+    try!((&mut buf[8..]).write_all(RECORD_PREFIX_SEP));
+    Ok(())
+}
+
+fn append_table_index_prefix(mut buf: &mut [u8], table_id: i64) -> Result<()> {
+    try!(buf.write_all(TABLE_PREFIX));
+    try!(number::encode_i64(buf, table_id));
+    try!((&mut buf[8..]).write_all(INDEX_PREFIX_SEP));
+    Ok(())
+}
+
+/// `encode_row_key` encodes the table id and record handle into a byte array.
+pub fn encode_row_key(table_id: i64, encoded_handle: &[u8]) -> Result<Vec<u8>> {
+    let mut k = vec![0; RECORD_ROW_KEY_LEN];
```

Comment:
`k` seems like a number variable like `i` or `j`, so why not using `key` instead?